src/App.tsx

```diff
@@ -1,26 +1,289 @@
+      case Filter.ACTIVE:
+        return !todo.completed;
+      case Filter.COMPLETED:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={`todoapp__toggle-all ${todos.every(todo => todo.completed) ? 'active' : ''}`}
+            data-cy="ToggleAllButton"
+          />
+
+          <form
+            onSubmit={e => {
+              e.preventDefault();
+              if (!loading) {
+                addTodo(newTodoTitle);
+              }
+            }}
+          >
+            <input
+              ref={inputRef}
+              name="todo"
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              value={newTodoTitle}
+              onChange={e => setNewTodoTitle(e.target.value)}
+              disabled={loading}
+            />
+          </form>
+        </header>
+
+        <section className="todoapp__main" data-cy="TodoList">
+          {filteredTodos.map(todo => (
+            <div
+              data-cy="Todo"
+              className={`todo ${todo.completed ? 'completed' : ''}`}
+              key={todo.id}
+            >
+              <label className="todo__status-label">
+                <input
+                  data-cy="TodoStatus"
+                  type="checkbox"
+                  className="todo__status"
+                  checked={todo.completed}
+                  onChange={() => updateTodoStatus(todo.id, !todo.completed)}
+                />
+              </label>
+              <span data-cy="TodoTitle" className="todo__title">
+                {todo.title}
+              </span>
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+                onClick={() => deleteTodo(todo.id)}
+              >
+                ×
+              </button>
+
+              <div
+                data-cy="TodoLoader"
+                className={`modal overlay ${loadingTodoId === todo.id ? 'is-active' : ''}`}
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          ))}
```

Comment:
```suggestion
          ))}
          {tempTodo && <TodoItem ....../>
```
and if tempTodo exists we will render this temp item